### PR TITLE
Openshift Virtualization Service Catalog Support

### PR DESCRIPTION
**Pull request by nasark**
Changes needed to support service catalog provisioning for the OSV provider

Depends on:
- [x] https://github.com/ManageIQ/manageiq-providers-kubevirt/pull/250

@miq-bot assign @agrare 
@miq-bot add_labels enhancement, radjabov/yes?
@miq-bot add_reviewer @agrare

### CODE REVIEW

File: app/models/manageiq/providers/openshift/infra_manager/provision_workflow.rb [hunk 1]
```diff
@@ -4,6 +4,6 @@ def self.provider_model
   end
 
   def dialog_name_from_automate(message = 'get_dialog_name')
-    super(message, {'platform' => 'openshift'})
+    super(message, "openshift")
```
Comment (nasark) [inline app/models/manageiq/providers/openshift/infra_manager/provision_workflow.rb:7]: `super` call here did not match the parameters of method in the kubevirt provider https://github.com/ManageIQ/manageiq-providers-kubevirt/pull/250/files#diff-ed8161fa01fc7bedf7e66a09c7a3b212b9c4452967bb912425c9e4ee1957a7ebR32

File: content/miq_dialogs/miq_provision_openshift_dialogs_template.yaml [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+            :order: []
+
+            :single_select: []
+
+            :exclude: []
+
+          :display: :edit
+          :required_tags: []
+
+          :data_type: :integer
```
Comment (agrare) [inline content/miq_dialogs/miq_provision_openshift_dialogs_template.yaml:133]: Realize this is just a copy&paste from kubevirt but I wonder why the odd spacing here...maybe follow-up to change it in both places later

Comment (nasark) [reply content/miq_dialogs/miq_provision_openshift_dialogs_template.yaml]: fixed

**Review by agrare — APPROVED** [review]
image
image

Successfully able to provision a VM from a template with normal lifecycle provisioning as well as from a service catalog item

**Comment by Fryguy** [comment]
Backported to `radjabov` in commit 7951cc8a854a17ae729ae3bbeef113fae35828e7.

```text
commit 7951cc8a854a17ae729ae3bbeef113fae35828e7
Author: Adam Grare <adam@grare.com>
Date:   Wed Sep 25 13:16:05 2024 -0400

    Merge pull request #266 from nasark/service_catalog_support
    
    Openshift Virtualization Service Catalog Support
    
    (cherry picked from commit 46e75f7defcedd1250beabc14d5a4a8255143c99)
```

**Pull request merged**